src/portal/lib/src/vulnerability-scanning/result-tip-histogram/result-tip-histogram.component.ts

```diff
@@ -0,0 +1,176 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { VulnerabilitySummary } from "../../service";
+import { VULNERABILITY_SCAN_STATUS, VULNERABILITY_SEVERITY } from "../../utils";
+import { TranslateService } from "@ngx-translate/core";
+
+@Component({
+  selector: 'hbr-result-tip-histogram',
+  templateUrl: './result-tip-histogram.component.html',
+  styleUrls: ['./result-tip-histogram.component.scss']
+})
+export class ResultTipHistogramComponent implements OnInit {
+  _tipTitle: string = "";
+  @Input() vulnerabilitySummary: VulnerabilitySummary = {
+    scan_status: VULNERABILITY_SCAN_STATUS.NOT_SCANNED,
+    severity: "",
+  };
+  constructor(private translate: TranslateService) { }
+
+  ngOnInit(): void {
+    let key = "VULNERABILITY.SEVERITY.UNKNOWN";
+    switch (this.vulnerabilitySummary.severity) {
+      case VULNERABILITY_SEVERITY.CRITICAL:
+          key = "VULNERABILITY.SEVERITY.CRITICAL";
+          break;
+      case VULNERABILITY_SEVERITY.HIGH:
+          key = "VULNERABILITY.SEVERITY.HIGH";
+          break;
+      case VULNERABILITY_SEVERITY.MEDIUM:
+          key = "VULNERABILITY.SEVERITY.MEDIUM";
+          break;
+      case VULNERABILITY_SEVERITY.LOW:
+          key = "VULNERABILITY.SEVERITY.LOW";
+          break;
+      case VULNERABILITY_SEVERITY.NEGLIGIBLE:
+          key = "VULNERABILITY.SEVERITY.NEGLIGIBLE";
+          break;
+      default:
+          break;
+    }
+
+    this.translate.get(key).subscribe( (res: string) => {
+      this._tipTitle = res;
+    });
+  }
+
+  get tipTitle(): string {
+    return this._tipTitle;
+  }
+
+  get total(): number {
+    if (this.vulnerabilitySummary &&
+      this.vulnerabilitySummary.summary) {
+        return this.vulnerabilitySummary.summary.total;
+    }
+
+    return 0;
+  }
+
+  get sevSummary(): {[key: string]: number} {
+    if (this.vulnerabilitySummary &&
+      this.vulnerabilitySummary.summary) {
+      return this.vulnerabilitySummary.summary.summary;
+    }
+
+    return null;
+  }
+
+  get criticalCount(): number {
+    if (this.sevSummary) {
+      return this.sevSummary[VULNERABILITY_SEVERITY.CRITICAL];
+    }
+
+    return 0;
+  }
+
+  get highCount(): number {
+    if (this.sevSummary) {
+      return this.sevSummary[VULNERABILITY_SEVERITY.HIGH];
+    }
+
+    return 0;
+  }
+
+  get mediumCount(): number {
+    if (this.sevSummary) {
+      return this.sevSummary[VULNERABILITY_SEVERITY.MEDIUM];
+    }
+
+    return 0;
+  }
+
+  get lowCount(): number {
+    if (this.sevSummary) {
+      return this.sevSummary[VULNERABILITY_SEVERITY.LOW];
+    }
+
+    return 0;
+  }
+
+  get unknownCount(): number {
```

Comment:
Excessive use of getter can affect performance

Reply:
Sure.